src/OrbitGl/TrackManager.cpp

```diff
@@ -103,12 +104,18 @@ void TrackManager::SortTracks() {
   }
 
   // Tracepoint tracks.
-  if (!tracepoints_system_wide_track_->IsEmpty()) {
-    all_processes_sorted_tracks.push_back(tracepoints_system_wide_track_);
+  if (absl::GetFlag(FLAGS_enable_tracepoint_feature)) {
+    if (app_ != nullptr && app_->HasCaptureData()) {
+      ThreadTrack* tracepoints_track =
+          GetOrCreateThreadTrack(orbit_base::kAllThreadsOfAllProcessesTid);
+      if (!tracepoints_track->IsEmpty()) {
+        all_processes_sorted_tracks.push_back(tracepoints_track);
+      }
+    }
   }
 
   // Process track.
-  if (app_->HasCaptureData()) {
+  if (app_ != nullptr && app_->HasCaptureData()) {
```

Comment:
What's the purpose of making app_ optional? For tests I assume?

Have you considered replacing the pointer to OrbitApp by a pointer to an interface which can be mocked for tests? Something like this for example:

```
class TrackAppInterface {
public:
  virtual bool HasCaptureData() const  = 0;
  virtual bool IsCapturing() const = 0;
};
```

Or is that supposed to be a temporary solution?

Reply:
As discussed: Mainly for tests. I wasn't sure if this is temporary, but we agree that it should be changed later. The Interface solution is a lot of effort for this change since it "bubbles down" to all tracks, and I don't want to go there just now...